### PR TITLE
Kelsha Darby - User Story 7

### LINKED ISSUE #10: User Story 7, User Navigation Restrictions
As a default user
When I try to access any path that begins with the following, then I see a 404 error:
- '/merchant'
- '/admin'

### PR DESCRIPTION
**Pull request by kelshadarby**
### Ensure regular users cannot view merchant/admin specific pages

_Test_
- Renamed `user_navigation_spec` to `user/nav_spec` and `merchant/nav_spec` for better clarity
- Added to `user/nav_bar_spec` to test that a default user is redirected to a 404 page if they try to access merchant and admin pages
- Fixed `default_user_navigation_spec` `click_on "View My Profile"` to match link name in application view

_Controller_
- Within `admin`
    - Created a `base_controller` that redirects any unauthorized users to a 404 page
    - Changed the `users_controller` and the `dashboard_controller` to inherit from the base controller
- Within `merchant`
    - Created a `base_controller` that redirects any unauthorized users to a 404 page
    - Changed the the `dashboard_controller` to inherit from the base controller
- Within `application_controller`
    - Added a `current_merchant?` helper method and removed the unnecessary `require_admin` from that controller

Closes #10

### CODE REVIEW

File: app/controllers/admin/base_controller.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Admin::BaseController < ApplicationController
+  before_action :require_admin
+
+   def require_admin
+     render file: "/public/404" unless current_admin?
```
Comment (fredrondina) [inline app/controllers/admin/base_controller.rb:5]: nice job getting this set up now

File: app/controllers/merchant/base_controller.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class Merchant::BaseController < ApplicationController
+  before_action :require_merchant
+
+   def require_merchant
+     render file: "/public/404" unless current_merchant?
```
Comment (fredrondina) [inline app/controllers/merchant/base_controller.rb:5]: nice verification

File: spec/features/users/default_user_navigation_spec.rb.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        city: "Userville",
+        state: "State 1",
+        zip_code: "12345"
+      )
+    )
+
+    allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(default_user)
+
+    visit "/merchant/dashboard"
+    expect(page).to have_content("The page you were looking for doesn't exist (404)")
```
Comment (fredrondina) [inline spec/features/users/default_user_navigation_spec.rb.rb]: nice tests

**Review by fredrondina — APPROVED** [review]
looks good to me, I am fine with the naming of the tests

**Review by fredrondina — APPROVED** [review]
nice changes

**Review by brian-greeson — APPROVED** [review]
This is great

**Pull request merged**